static/js/snap-metrics.js

```diff
@@ -0,0 +1,293 @@
+/* global d3, bb, moment */
+
+(function (window, d3, bb, moment) {
+  /**
+   * Debounce
+   * @param {Function} func Function to run.
+   * @param {Number} wait Time to wait between tries.
+   * @param {Boolean} immediate Immediately call func.
+   */
+  function debounce(func, wait, immediate) {
+    var timeout;
+    return function() {
+      var context = this, args = arguments;
+      var later = function() {
+        timeout = null;
+        if (!immediate) func.apply(context, args);
+      };
+      var callNow = immediate && !timeout;
+      clearTimeout(timeout);
+      timeout = setTimeout(later, wait);
+      if (callNow) func.apply(context, args);
+    };
+  }
+
+  /**
+   * Render all metrics
+   * @param {Object} metrics An object of metrics from the API.
+   */
+  function renderMetrics(metrics) {
+    if (!d3 || ! bb) {
+      return false;
+    }
+
+    var X_TICK_FREQUENCY = 7;
+    var Y_TICK_FREQUENCY = 5;
+
+    var COLORS = {
+      installs: '#94519E'
+    };
+
+    /**
+     * Cull Y Axis. The built in culling does not provide enough control.
+     *  - hide labels that are not every X_TICK_FREQUENCY ticks.
+     *  - remove month abreviation from label for sequential dates that have 
+     *    the same month.
+     * @param {NodeList} ticks X axis tick elements.
+     */
+    function cullXAxis(ticks) {
+      var tick, totalTicks, text, monthCache;
+
+      for (tick = 0, totalTicks = ticks.length; tick < totalTicks; tick += 1) {
```

Comment:
Unless you need a `tick` for anything else then iterator, this seems like it could be a `forEach` as well...